### DISCUSSION TITLE
BroadcastChannel.onMessage() runs multi-times. Why?

### DISCUSSION THREAD

**Discussion by sinyo-matu** [alephjs/aleph.js#500] [Q&A] [answered]
Hi everyone.

I write BroadcastChannel.onMessage(handler) in my server.ts file.
And I expect the handler runs once on every message incoming.

But when an other BroadcastChannel instance (constructed in a route request hander) send one message,
the handler passed in onMessage() method runs at least three times.
And it will increase by 2 times by every hot reload.

Anybody know why this behavior happen?

**Comment by sinyo-matu** [comment]
It seems like this behavior is caused by dev mode.
When I ran `deno task start` this behavior disappeared!

**Comment by ije** [comment, marked as answer]
dev will re-import the server.ts with random query when it changes, the 'side effect' wouldn't be removed automatically
